text/0016-auto-code-mappings.md

```diff
@@ -0,0 +1,86 @@
+# Automatic code mappings
+
+* Start Date: 2022-09-26
+* RFC Type: decision
+* RFC PR: <https://github.com/getsentry/rfcs/pull/16>
+
+## Summary
+
+We are looking into creating code mappings automatically for organizations that have code source management installed.
+
+We will initially target orgs with Github installations with Python and JS projects (when source maps are available) that do *not* have code mappings already in place.
+Other source code management integrations and projects will be considered in the future.
+
+## Motivation
+
+There are various features not available to customers when code mappings are not in place for projects that have stacktraces (see background section for more details). Reducing the number of projects without code mappings will increase the value provided to customers. Those features are also required in order to reduce the "notify everyone" problem that Sentry currently suffers from.
+
+## Background
+
+Sentry wants developers to have context in issue details that allows them to easily identify and fix issues. Context includes the following:
+
+* Code Context
+  * Stack Trace Links ([link](https://docs.sentry.io/product/integrations/source-code-mgmt/github/#stack-trace-linking) to docs)
+  * Commit Context (Suspect Commits) ([link](https://docs.sentry.io/product/releases/suspect-commits/) to docs)
+* Assignee Context
+  * Code owners/Ownership rules
+
+Properly set up code mappings is required in order to enable the proper functioning of the features listed above.
+
+## Supporting Data
+
+About 20% of repos connected to Sentry have code mappings and very few have both the stack trace and source code root values set which can lead to improper stack trace linking and code assignments. For complete data follow [link for staff](https://www.notion.so/sentry/Deriving-Code-Mappings-9086faaf3fed4faca69e8b35f8f70e26#bae13a87214d4f52b936e0e1aa6829ec).
+
+## Details
+
+A code mapping allows going from a stackframe module path to a source file in the preferred source code management tool:
+
+<img src="0016-code-mapping-logic.png" width="500px" />
+
+The process is quite simple. Give a stacktrace frame, we look for the file name in all of the repos we have access for a customer's Github org. For instance, given the stacktrace path `sentry/integrations/gitlab/client.py` look for `integrations/gitlab/client.py` and if there is a unique match we have determined all the values for the code mapping. For the curious, a rudimentary POC (only for Python atm) can be viewed and tested [here](https://gist.github.com/armenzg/40ba48fff217815842c4fe16047d0835).
```

Comment:
Should we be explicit about this *not* working for java as file paths are not always present?